Bottom: 8e92928fe5613c705f3b9facd7e6b85f724145e9
Top:    ccf1c3ae432b3ce1789185552df6b20b965cdd01
Author: Parth Shah <parth@linux.ibm.com>
Date:   2020-01-31 12:23:30 +0530

All callers of available_idle_cpu are in kernel/sched/fair.c
So move the function definition to fair.c

No functional change.

Signed-off-by: Parth Shah <parth@linux.ibm.com>
Signed-off-by: Srikar Dronamraju <srikar@linux.vnet.ibm.com>


---

diff --git a/kernel/sched/core.c b/kernel/sched/core.c
index fc1dfc007604..30704f2037db 100644
--- a/kernel/sched/core.c
+++ b/kernel/sched/core.c
@@ -4642,23 +4642,6 @@ int idle_cpu(int cpu)
 	return 1;
 }
 
-/**
- * available_idle_cpu - is a given CPU idle for enqueuing work.
- * @cpu: the CPU in question.
- *
- * Return: 1 if the CPU is currently idle. 0 otherwise.
- */
-int available_idle_cpu(int cpu)
-{
-	if (!idle_cpu(cpu))
-		return 0;
-
-	if (vcpu_is_preempted(cpu))
-		return 0;
-
-	return 1;
-}
-
 /**
  * idle_task - return the idle task for a given CPU.
  * @cpu: the processor in question.
diff --git a/kernel/sched/fair.c b/kernel/sched/fair.c
index 6690845f4d79..36599b6155ea 100644
--- a/kernel/sched/fair.c
+++ b/kernel/sched/fair.c
@@ -5453,6 +5453,23 @@ static unsigned long capacity_of(int cpu)
 	return cpu_rq(cpu)->cpu_capacity;
 }
 
+/*
+ * available_idle_cpu - is a given CPU idle for enqueuing work.
+ * @cpu: the CPU in question.
+ *
+ * Return: 1 if the CPU is currently idle. 0 otherwise.
+ */
+int available_idle_cpu(int cpu)
+{
+       if (!idle_cpu(cpu))
+               return 0;
+
+       if (vcpu_is_preempted(cpu))
+               return 0;
+
+       return 1;
+}
+
 static void record_wakee(struct task_struct *p)
 {
 	/*
